agent/src/api/v1/storage.rs

```diff
@@ -11,8 +13,53 @@ use utils::structs::Manifest;
 
 use crate::structures::*;
 
+/// Mount all storage endpoint handlers onto the passed-in router.
+pub fn mount(router: ServalRouter) -> ServalRouter {
+    router
+        .route("/v1/storage/manifests", get(list_manifests))
+        .route("/v1/storage/manifests", post(store_manifest))
+        .route("/v1/storage/manifests/:name", get(get_manifest))
+        .route("/v1/storage/manifests/:name", head(has_manifest))
+        .route(
+            "/v1/storage/manifests/:name/executable/:version",
+            put(store_executable),
+        )
+        .route(
+            "/v1/storage/manifests/:name/executable/:version",
+            get(get_executable),
+        )
+}
+
+/// Relay all storage requests to a node that can handle them.
+pub async fn proxy(
+    State(state): State<AppState>,
+    mut req: Request<Body>,
+    next: Next<Body>,
+) -> Result<Response, StatusCode> {
+    // TODO: I kinda don't like this as middleware, but maybe it's the cleanest way to do it.
```

Comment:
Are there any other alternatives that are any better than putting something at the top of each individual handler? We could write that so it wouldn't be particularly hideous, but it's still boilerplate:

```rust
async fn get_executable(
     Path((name, version)): Path<(String, String)>,
     State(_state): State<AppState>,
 ) -> impl IntoResponse {

    if let Some(proxy_req) = maybe_proxy_request(&mut req, &state) {
        let resp = proxy_req.await;
        return Ok(resp);
    }

    // ... rest of handler goes here
````